src/tree/workspace/sharedWorkspaceStorage.ts

```diff
@@ -0,0 +1,160 @@
+ */
+export class SharedWorkspaceStorage {
+    private static readonly storageName: string = 'ms-azuretools.vscode-cosmosdb.workspace';
+
+    /**
+     * Retrieves all items from the storage along with their secrets.
+     * The item's `id` is used as the key and must be unique per `workspaceType`.
+     *
+     * @param workspaceType - The type of the workspace resource.
+     * @returns An array of items.
+     */
+    public static async getItems(workspaceType: WorkspaceResourceType): Promise<SharedWorkspaceStorageItem[]> {
+        const storageKeyPrefix = `${SharedWorkspaceStorage.storageName}/${workspaceType}/`;
+        const keys = ext.context.globalState.keys().filter((key) => key.startsWith(storageKeyPrefix));
+        const items: SharedWorkspaceStorageItem[] = [];
+
+        for (const key of keys) {
+            const item = ext.context.globalState.get<SharedWorkspaceStorageItem>(key);
+            if (item) {
+                // Read secrets associated with the item
+                const secretKey = `${key}/secrets`;
+                const secretsJson = await ext.secretStorage.get(secretKey);
+
+                let secrets: string[] = [];
+                if (secretsJson) {
+                    try {
+                        secrets = JSON.parse(secretsJson) as string[];
+                    } catch (error) {
+                        console.error(`Failed to parse secrets for key ${key}:`, error);
+                        secrets = [];
+                    }
+                }
+
+                item.secrets = secrets;
+                items.push(item);
+            }
+        }
+
+        return items;
+    }
+
+    /**
+     * Stores an item and its secrets into the global state and secret storage.
+     * The item's `id` is used as the key and must be unique per `workspaceType`.
+     *
+     * @param workspaceType - The type of the workspace resource.
+     * @param item - The item to store.
+     * @param overwrite - If `false` and an item with the same `id` exists, an error is thrown.
+     *                    Defaults to `true`.
+     * @throws Error if `overwrite` is `false` and an item with the same `id` exists.
+     */
+    public static async push(
+        workspaceType: WorkspaceResourceType,
+        item: SharedWorkspaceStorageItem,
+        overwrite: boolean = true,
+    ): Promise<void> {
+        const storageKey = `${SharedWorkspaceStorage.storageName}/${workspaceType}/${item.id}`;
+
+        // Check for existing item
+        const existingItem = ext.context.globalState.get<SharedWorkspaceStorageItem>(storageKey);
+        if (existingItem && !overwrite) {
+            throw new Error(`An item with id '${item.id}' already exists for workspaceType '${workspaceType}'.`);
+        }
+
+        // Save all secrets
+        if (item.secrets && item.secrets.length > 0) {
+            const secretKey = `${storageKey}/secrets`;
```

Comment:
Maybe let's use the existing [`getSecretStorageKey`](https://github.com/microsoft/vscode-cosmosdb/blob/main/src/utils/getSecretStorageKey.ts#L7) func? And probably reuse the [existing key for attached accounts](https://github.com/microsoft/vscode-cosmosdb/blob/main/src/tree/AttachedAccountsTreeItem.ts#L56) so all creds are stored in the same place.

Reply:
I looked into this while introducing the new `SharedWorkspaceStorage`.
Here, I reused the prefix `ms-azuretools.vscode-cosmosdb` and added `workspace` to the namespace:
```ts
    private static readonly storageName: string = 'ms-azuretools.vscode-cosmosdb.workspace';
```
This approach introduces a new namespace structure to support the storage of API-specific data and secrets beyond connection string, i.e. we might store additional keys/passwords, etc.. For example, I introduced the `secrets` branch instead of a `connectionString` branch.

Each workspace branch data provider is assigned its own namespace, nested under the common prefix.

The goal is to start with a unified but extensible structure. I’d like to migrate the existing "Attached Database Accounts" to this new structure as part of the V1 to V2 migration.